### PR TITLE
Shipping billing addresses feature

### LINKED ISSUE #70: [Feature Request] Customer Adress (Shipping / Billing)
Thank you for open wc-smooth-generator to the public. We've integrated this to test our extensions for Woocommerce better and make them more reliable.

**Is your feature request related to a problem? Please describe.**
Currently created orders are containing only basic data. Customers always use the same billing and shipping adress when orders are generated.

**Describe the solution you'd like**
I would be helpfull if the customers created dont have always the same shipping and billing adress. At some customers, the shipping and billing adresses should be different to better reflect the real world cases.

We understand that this repo is there to easily gather some testdata and not have a complete set of each use-case that would be possible in woo. Still we believe this could be interesting for a wider audience.

### PR DESCRIPTION
**Pull request by michi-dev**
### All Submissions:

* [x] Have you followed the [Contributing guidelines](https://github.com/woocommerce/wc-smooth-generator/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/wc-smooth-generator/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #70. 

I have added more functionality than described in the issue #70. This needs a bit further explanation. I have divorced any shipping / billing address in three parts:
- person (firstname, lastname)
- address (address1,address2, city, state, postcode, country)
- company (company_name)

Every customer, who gets generated, has those three parts. Every part has a random generation. I added the feature that at the customer generation in Customer.php:
possibility A: billing = shipping (was done before)
possibility B: 1/2 billing != shipping (added with this pull request)

therefore, the shipping address can have different details for
- personal information (other person / same person)
- location information (other location / same location)
- company name (B2B (same company), B2B (different companies), B2C, C2B, C2C)

every part has its own random generation. with all three parts coming together its a unique order with in total (2*2*5) 40 combinations and therefore test cases.

### How to test the changes in this Pull Request:

1.use `wp wc generate customers 50` to directly test the customer generation 
2.use `wp wc generate orders 50` to see if the customers were correctly saved to the orders

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: includes/Generator/Customer.php [hunk 1]
```diff
@@ -30,50 +31,109 @@ public static function generate( $save = true ) {
...
-			'shipping_state'      => $state,
-			'shipping_postcode'   => $postcode,
-			'shipping_country'    => $countrycode,
-			'is_paying_customer'  => false,
-		) );
+		/*PERSON*/
+		$person['billing']['firstname'] = self::$faker->firstName( self::$faker->randomElement( array( 'male', 'female' ) ) );
+		$person['billing']['lastname']  = self::$faker->lastName();
+
+		if ( self::$faker->randomFloat( 0, 0, 1 ) == 1 ) {
```
Comment (Luc45) [inline includes/Generator/Customer.php]:
```suggestion
		// 50% chance
		if ( (bool) wp_rand( 0, 1 ) ) {
```

File: includes/Generator/Customer.php [hunk 2]
```diff
@@ -30,50 +31,109 @@ public static function generate( $save = true ) {
...
+		$address['billing']['address0'] = self::$faker->buildingNumber() . ' ' . self::$faker->streetName();
+		$address['billing']['address1'] = self::$faker->streetAddress();
+		$address['billing']['city']     = self::$faker->city();
+		$address['billing']['state']    = self::$faker->stateAbbr();
+		$address['billing']['postcode'] = self::$faker->postcode();
+		$address['billing']['country']  = self::$faker->countryCode();
+		$address['billing']['phone']    = self::$faker->e164PhoneNumber();
+		$address['billing']['email']    = $email;
+
+		if ( self::$faker->randomFloat( 0, 0, 1 ) == 1 ) {
```
Comment (Luc45) [inline includes/Generator/Customer.php]:
```suggestion
		// 50% chance
		if ( (bool) wp_rand( 0, 1 ) ) {
```

**Review by Luc45 — DISMISSED** [review]
Works well, thanks for contributing!

I have left just some minor readability suggestions

I'll forward this internally to be merged

image

**Comment by michi-dev** [comment]
No problem, glad to see it worked @Luc45

**Comment by masteradhoc** [comment]
@samueljseay can u check and merge this one also?

**Comment by jonathansadowski** [comment]
Hi @michi-dev,

Apologies not getting to this sooner. It looks like there are some conflicts now. Would you mind resolving those so that we can give this a re-review?

**Comment by samueljseay** [comment]
It looks like there are still some conflicts here @michi-dev just bumping this in case you missed it. Sorry again for the delayed response on my part.

**Comment by michi-dev** [comment]
@samueljseay @jonathansadowski should now be fine - can you check?

**Comment by jonathansadowski** [comment]
Hi @michi-dev,

Sorry for the delay in getting back to this. Looks great, thanks for your contribution!

**Review by jonathansadowski — APPROVED** [review]

**Pull request merged**